lib/vagrant-service-manager/service.rb

```diff
@@ -0,0 +1,35 @@
+# Loads all services
+Dir["#{File.dirname(__FILE__)}/services/*.rb"].each { |f| require_relative f }
+
+module Vagrant
+  module ServiceManager
+    SUPPORTED_BOXES = ['adb', 'cdk']
+
+    class Service
+      def initialize(app, env)
+        @app = app
+        @machine = env[:machine]
+        @ui = env[:ui]
+        @services = @machine.config.servicemanager.services.split(',').map(&:strip)
+        @docker_hook = Docker.new(@machine, @ui)
+        @openshift_hook = OpenShift.new(@machine, @ui)
+      end
+
+      def call(env)
+        @app.call(env)
+
+        if SUPPORTED_BOXES.include? @machine.guest.capability(:os_variant)
+          @docker_hook.execute
+
+          if @machine.guest.capability(:os_variant) == "cdk"
```

Comment:
> We should only execute here if there is no services line. We should not force openshift on CDK, only run it as default.

The only two supported services we can configure for user is `docker` and `openshift`. `docker` service is configured by default irrespective of the box in action, which leaves us to `openshift`. And since `openshift` to be enabled by default for CDK we are enabling it by default irrespective of the explicit mention in Vagrantfile (as for docker).

Later, when we'll have support for more services (kubernetes, mesos) the logic should be expanded to handle the collision and check for explicit mention of the services.

Given the scope of feature we have for this release, I think this solves the scoped problem in hand for the release.